bolt/base.go

```diff
@@ -0,0 +1,89 @@
+package bolt
+
+import (
+	"github.com/boltdb/bolt"
+)
+
+// SchemaVersionBucket stores ids of completed migrations
+var SchemaVersionBucket = []byte("SchemaVersions")
+
+// IsMigrationComplete checks for the presence of a particular migration id
+func IsMigrationComplete(db *bolt.DB, id string) (bool, error) {
+	complete := false
+	if err := db.View(func(tx *bolt.Tx) error {
+		migration := tx.Bucket(SchemaVersionBucket).Get([]byte(id))
+		if migration != nil {
+			complete = true
+		}
+		return nil
+	}); err != nil {
+		return true, err
+	}
+
+	return complete, nil
+}
+
+// MarkMigrationAsComplete adds the migration id to the schema bucket
+func MarkMigrationAsComplete(db *bolt.DB, id string) error {
```

Comment:
Just a thought, do you think it might be worth it to store the datetime when the migration was applied. Could be helpful debugging information if something goes awry.

Reply:
Good idea.